torch/distributed/device_mesh.py

```diff
@@ -560,17 +560,19 @@ def _init_process_groups(self):
                     # numbers of API calls are equal to the number of subgroups for each mesh dimension. In a 2 * 4
                     # mesh, we need to make 2 + 4 = 6 API calls per ranks to create all the subgroups.
                     dim_group = None
+                    has_split_group = False
                     if (
                         bound_device_id := getattr(
                             default_group, "bound_device_id", None
                         )
-                    ) is not None:
+                    ) is not None and torch.cuda.is_available():
```

Comment:
Doens't the mesh has a device type concept? You might be running on a machine that has cuda build building a cpu mesh here right?

Reply:
> Doens't the mesh has a device type concept? 

I think so. From current PyTorch design, device type is a request for `DeviceMesh`. https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L379 

> You might be running on a machine that has cuda build building a cpu mesh here right?

I don't try that but it seems to be a valid case. `cpu` could be the device_type of `DeviceMesh`  from the comments https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L397


What I change in this PR is limit `split_group` for CUDA device only as PyTorch `split_group` only works for NCCL backend. https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L4687